mitwelten/mitwelten-db-backend
# 48 rename entries to notes

- [x] rename tables, models and endpoints
- [x] make location of notes optional
- [x] add public flag
- [x] change field name 'name' to 'title'
- [x] add field for KeyCloak user_sub

### Requires changes to:

- [x] deploy.mitwelten.org (tags list)
- [ ] walk.mitwelten.org (forms, etc.)
- [ ] discover.mitwelten.org

### Testing:

on the database server:

- clone the repo onto the database server and check out this branch:
    ```bash
    git clone https://github.com/mitwelten/mitwelten-db-backend.git
    cd mitwelten-db-backend
    git checkout origin/48-rename-entries-to-notes
    ```
- create `credentials.py` with `mitwelten_admin` as database user
- run the desired migration procedure (see [NOTES.md](https://github.com/mitwelten/mitwelten-db-backend/blob/d23c4efada5cd8ea8c192fa905f9b28f5e1a9272/schema/NOTES.md#migration-v23-to-v24))

on your machine:

- clone the repo onto your machine and check out this branch (see above)
- create `credentials.py` with `mitwelten_internal` as database user and `dev` as schema
- activate and install venv:
    ```bash
    cd services/api
    python -m venv .venv
    source .venv/bin/activate
    pip install -r requirements
    ```
- run the REST API service: 
    ```bash
    cd services
    ./api/.venv/bin/uvicorn api.main:app --reload
    ```

### Discussions:

There are viz-app specific endpoints for entries, nodes, etc. I'd appreciate a simplification and would like to remove those endpoints. Does the discover app rely on those endpoints? PR #50 implements this simplification.
